v2/internal/testcommon/error_translating_roundtripper_v1.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright (c) Microsoft Corporation.
```

Comment:
I'll give this a go.

Reply:
This worked out well, though there were some cascading effects. I used the package name `vcr` to avoid clashing with existing names like `recorder` and `record`.